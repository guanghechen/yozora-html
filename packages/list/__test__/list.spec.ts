import type { List } from '@yozora/ast'
import { renderChildren } from '../../../jest.setup'
import renderList from '../src'

describe('snapshot', function () {
  it('basic', function () {
    const node = {
      type: 'list',
      ordered: false,
      marker: 45,
      spread: true,
      children: [
        {
          type: 'listItem',
          children: [
            {
              type: 'paragraph',
              children: [
                {
                  type: 'text',
                  value: 'a',
                },
              ],
            },
          ],
        },
        {
          type: 'listItem',
          children: [
            {
              type: 'paragraph',
              children: [
                {
                  type: 'text',
                  value: 'b',
                },
              ],
            },
          ],
        },
      ],
    }
    expect(renderList(node as List, renderChildren)).toMatchSnapshot()
  })

  it('todo', function () {
    const node = {
      type: 'list',
      ordered: false,
      marker: 45,
      spread: false,
      children: [
        {
          type: 'listItem',
          status: 'todo',
          children: [
            {
              type: 'text',
              value: 'foo',
            },
          ],
        },
        {
          type: 'listItem',
          status: 'done',
          children: [
            {
              type: 'text',
              value: 'bar',
            },
          ],
        },
      ],
    }
    expect(renderList(node as List, renderChildren)).toMatchSnapshot()
  })

  it('ordered', function () {
    const node = {
      type: 'list',
      ordered: true,
      start: 1,
      marker: 46,
      spread: false,
      children: [
        {
          type: 'listItem',
          children: [
            {
              type: 'text',
              value: 'a',
            },
          ],
        },
        {
          type: 'listItem',
          children: [
            {
              type: 'text',
              value: 'b',
            },
          ],
        },
        {
          type: 'listItem',
          children: [
            {
              type: 'text',
              value: 'c',
            },
          ],
        },
      ],
    }
    expect(renderList(node as List, renderChildren)).toMatchSnapshot()
  })
})
