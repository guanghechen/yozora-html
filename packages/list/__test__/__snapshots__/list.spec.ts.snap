// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `"<ul class="yozora-list"><li class="yozora-list-item"><p class="yozora-paragraph"><span class="yozora-text">a</span></p></li><li class="yozora-list-item"><p class="yozora-paragraph"><span class="yozora-text">b</span></p></li></ul>"`;

exports[`snapshot ordered 1`] = `"<ol class="yozora-list" start="1"><li class="yozora-list-item"><span class="yozora-text">a</span></li><li class="yozora-list-item"><span class="yozora-text">b</span></li><li class="yozora-list-item"><span class="yozora-text">c</span></li></ol>"`;

exports[`snapshot todo 1`] = `"<ul class="yozora-list"><li class="yozora-list-item"><input disabled="" type="checkbox" /> <span class="yozora-text">foo</span></li><li class="yozora-list-item"><input checked="" disabled="" type="checkbox" /> <span class="yozora-text">bar</span></li></ul>"`;
